halo/cmd/register_to_avs.go

```diff
@@ -0,0 +1,166 @@
+			cfg.HaloConfig.HomeDir = cfg.HomeDir
+
+			// read the comet config based on the home directory
+			cometCfg, err := parseCometConfig(cmd.Context(), cfg.HaloConfig.HomeDir)
+			if err != nil {
+				return err
+			}
+			cfg.CometConfig = cometCfg
+
+			return register(cmd.Context(), cfg)
+		},
+	}
+
+	bindOperatorFlags(registerToAVSCmd.Flags(), cfg)
+
+	return registerToAVSCmd
+}
+
+func register(ctx context.Context, cfg *OperatorConfig) error {
+	// check for home directory where the config files exist
+	if !directoryExists(cfg.HaloConfig.HomeDir) {
+		log.Info(ctx, "Make sure to run \"init\" command before running operator commands")
+		err := errors.New("directory does not exists", "home", cfg.HaloConfig.HomeDir)
+
+		return err
+	}
+
+	// load network config for the layer1 chain
+	chain, err := getChainConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	// load a private validator key and state from disk (this hard exits on any error).
+	privVal := privval.LoadFilePVEmptyState(cfg.CometConfig.PrivValidatorKeyFile(), cfg.CometConfig.PrivValidatorStateFile())
+
+	// connect to the rpc endpoint
+	client, err := ethclient.Dial(chain.Name, chain.RPCURL)
+	if err != nil {
+		return err
+	}
+
+	// load contract bindings
+	omniAvs, err := bindings.NewOmniAVS(common.HexToAddress(cfg.AVSDirectoryAddr), client)
+	if err != nil {
+		return err
+	}
+	avsDirectory, err := bindings.NewAVSDirectory(common.HexToAddress(cfg.OmniAVSAddr), client)
+	if err != nil {
+		return err
+	}
+
+	operPK, err := crypto.ToECDSA(privVal.Key.PrivKey.Bytes())
+	if err != nil {
+		log.Info(ctx, "Could not convert private keys", err)
+
+		return errors.Wrap(err, "could not convert pk to ecdsa")
+	}
+	operAddr := common.HexToAddress(privVal.GetAddress().String())
+
+	// calculate operator signature and digest hash
+	blockNumber, err := client.BlockNumber(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting blockNumber ")
+	}
+	block, err := client.BlockByNumber(ctx, big.NewInt(int64(blockNumber)))
+	if err != nil {
+		return errors.Wrap(err, "getting blockByNumber ")
+	}
+
+	operatorSignatureWithSaltAndExpiry := bindings.ISignatureUtilsSignatureWithSaltAndExpiry{
+		Signature: []byte{0},
+		Salt:      crypto.Keccak256Hash(operAddr.Bytes()),
+		Expiry:    big.NewInt(int64(block.Time()) + int64(24*time.Hour)),
+	}
+	digestHash, err := avsDirectory.CalculateOperatorAVSRegistrationDigestHash(&bind.CallOpts{},
+		common.HexToAddress(privVal.GetAddress().String()),
+		common.HexToAddress(cfg.OmniAVSAddr),
+		operatorSignatureWithSaltAndExpiry.Salt,
+		operatorSignatureWithSaltAndExpiry.Expiry)
+	if err != nil {
+		return err
+	}
+
+	operatorSignatureWithSaltAndExpiry.Signature, err = crypto.Sign(digestHash[:32], operPK)
+	if err != nil {
+		return errors.Wrap(err, "error signing)")
+	}
+
+	if len(operatorSignatureWithSaltAndExpiry.Signature) != 65 {
+		return errors.New("invalid signature length")
+	}
+	operatorSignatureWithSaltAndExpiry.Signature[64] += 27
```

Comment:
I think we have some utilities for validators to sign things in `lib/k1util` - maybe use them here?